test/test_index.py

```diff
@@ -555,12 +535,9 @@ def _count_existing(self, repo, files):
     # END num existing helper
 
     @pytest.mark.xfail(
-        os.name == "nt",
-        reason=(
-            "IndexFile.from_tree is broken on Windows (related to NamedTemporaryFile), see #1630.\n"
-            "'git read-tree --index-output=...' fails with 'fatal: unable to write new index file'."
-        ),
-        raises=GitCommandError,
+        os.name == "nt" and Git().config("core.symlinks") == "true",
```

Comment:
I'm unsure if this is the best way for the condition to check the `core.symlinks` configuration variable. This will typically run with the CWD as the GitPython repository, thereby including the local scope, and maybe only the system and global scopes should be used.

Reply:
I think GitPython has the same problem that `gitoxide` *would* have: It fails to see the configuration coming with the git installation itself, which has to be discovered [by asking `git` itself](https://github.com/Byron/gitoxide/blob/bc44497606656cddc4f18a0acb272c34b8df4ba8/gix-path/src/env/git.rs#L10).

That's the configuration file that contains these values, on Windows at least. Maybe Windows will also put that value into the repository-local configuration file, which might be the reason this condition works at all.

And since it works, I suppose it's good enough for now?

Reply:
> Maybe Windows will also put that value into the repository-local configuration file, which might be the reason this condition works at all.

At least in the more common `false` case, I've found that it is coped into the local repository's configuration. But using `Git().config(...)` doesn't rely on that. It works outside any repository and finds the system configuration:

```text
(.venv) C:\Users\ek\source\repos\GitPython [fromtree ≡]> pushd ~/tmp
(.venv) C:\Users\ek\tmp> python
Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import git
>>> git.Git().config("core.symlinks", show_origin=True)
'file:C:/Users/ek/scoop/apps/git/2.43.0/etc/gitconfig\tfalse'
```

(This is the same GitPython as when I run it from the `GitPython` directory, since the virtual environment is activated and the package is installed in it using `pip install -e .`.)

So I think the condition is working because `config` is not being treated specially--unlike if `GitConfigParser` is used, the command in that condition it really is running `git config`.

```text
(.venv) C:\Users\ek\tmp> python
Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> logging.basicConfig(level=logging.INFO)
>>> import git
>>> git.Git.GIT_PYTHON_TRACE = "full"
>>> git.Git().config("core.symlinks")
INFO:git.cmd:git config core.symlinks -> 0; stdout: 'false'
'false'
```